app/src/main/java/net/geekinpurple/www/appofourown/Work.java

```diff
@@ -0,0 +1,168 @@
+    protected Category category;
+    protected Warnings warnings;
+    protected Status status;
+
+    public Work(Element work) {
+        Elements workLinks = work.getElementsByTag("a");
+        this.title = workLinks.get(0).text(); // work title
+
+        String urlString = MainActivity.homeUrl;
+        StringBuffer url = new StringBuffer(urlString);
+        this.workUrl = url.append(workLinks.get(0).attr("href")).toString();
+        this.author = workLinks.get(1).text();
+        this.authorUrl = workLinks.get(1).attr("href");
+
+        String rate = work.getElementsByClass("rating").get(0).text();
+        switch (rate) {
+            case "General Audiences":
+                this.rating = Rating.G;
+                break;
+            case "Teen And Up Audiences":
+                this.rating = Rating.T;
+                break;
+            case "Mature":
+                this.rating = Rating.M;
+                break;
+            case "Explicit: only suitable for adults":
+                this.rating = Rating.E;
+                break;
+            default:
+                this.rating = Rating.N;
+                break;
+        }
+
+        String cat = work.getElementsByClass("category").get(0).text();
+        switch (cat) {
+            case "F/F":
+                this.category = Category.FF;
+                break;
+            case "F/M":
+                this.category = Category.FM;
+                break;
+            case "M/M":
+                this.category = Category.MM;
+                break;
+            case "Gen":
+                this.category = Category.GEN;
+                break;
+            case "No category":
+                this.category = Category.NONE;
+                break;
+            case "Other":
+                this.category = Category.OTHER;
+                break;
+            default:
+                this.category = Category.MULTI;
+                break;
+        }
+
+        this.warnings = Warnings.NONE;
+        this.status = Status.UNKNOWN;
+
+        Element desc = work.getElementsByClass("summary").get(0);
+        this.summary = desc.text();
+    }
+
+    // Reconstruct the Work from the Parcel
+    public Work(Parcel pkg) {
+        workUrl = pkg.readString();
+        title = pkg.readString();
+        author = pkg.readString();
+        authorUrl = pkg.readString();
+        summary = pkg.readString();
+
+        String temp = pkg.readString();
+        rating = Rating.valueOf(temp);
+        Log.d("testing", rating.toString());
+
+        category = Category.valueOf(pkg.readString());
+        Log.d("testing", category.toString());
+
+        temp = pkg.readString();
+        warnings = Warnings.valueOf(temp);
+        temp = pkg.readString();
+        status = Status.valueOf(temp);
+    }
+
+    protected int getRatingColor()
+    {
+        // {G, T, M, E, N}
+        switch (rating)
+        {
+            case G:
+                return 0xFF77a100;
```

Comment:
these are the colors used officially by AO3 http://archiveofourown.org/help/symbols-key.html